### PR TITLE
fix(config): Allow empty string as ref fallback

### PR DESCRIPTION
**Pull request by patrick96**
There is also a lot of refactoring that could be done here but since we are rewriting parts of the config parser already we'll do it there.

### CODE REVIEW

File: include/components/config.hpp [hunk 1]
```diff
@@ -268,17 +268,23 @@ class config {
...
+     */
+    bool has_default = false;
 
     if ((pos = var.find(':')) != string::npos) {
       env_default = var.substr(pos + 1);
+      has_default = true;
       var.erase(pos);
     }
 
     if (env_util::has(var.c_str())) {
```
Comment (Lomadriel) [inline include/components/config.hpp]:
```suggestion
    if (env_util::has(var)) {
```

File: include/components/config.hpp [hunk 2]
```diff
@@ -268,17 +268,23 @@ class config {
...
       string env_value{env_util::get(var.c_str())};
```
Comment (Lomadriel) [inline include/components/config.hpp]:
```suggestion
      string env_value{env_util::get(var)};
```

**Comment by patrick96** [comment]
> I also have a question, does it make sense to authorize empty file name?

Not really, why?

**Comment by Lomadriel** [comment]
Mostly because of your modification in `dereference_file` but I think it makes sense to be consistent and allow empty fallback everywhere.

**Comment by patrick96** [comment]
The fallback in `dereference_file` is an actual value for when the file isn't found not another path.

**Comment by Lomadriel** [comment]
> The fallback in `dereference_file` is an actual value for when the file isn't found not another path.

:man_facepalming: Indeed... (sorry)

**Pull request merged**